### DISCUSSION TITLE
Using ESP8266 Level Shifter Module instead OEM Wifi Dongle?

### DISCUSSION THREAD

**Discussion by MmeHulot** [syssi/esphome-soyosource-gtn-virtual-meter#98] [Q&A] [answered]
Hi ,
syssi mentioned using a ESP82 level shifter module instead of the OEM Wifi Dongle:
[https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/issues/50#issuecomment-1169189640](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/issues/50#issuecomment-1169189640)
Is it possible to use such a module, flash it with esp8266-wifi-dongle-example-web-only.yaml-Configuration  and then connect to the Soyo USB/TTL port or does only the original Wifi-Dongle work?
[https://de.aliexpress.com/item/32848420391.html](https://de.aliexpress.com/item/32848420391.html)
I would like to keep the OEM Dongle and replace it.

Thanks!

**Comment by syssi** [comment, marked as answer]
The project provides three different (poorly named) examples:

1. [esp8266-example.yaml](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/esp8266-example.yaml): Intended for the Soyosource with a display and a chatty RS485 which provides feedback to the `status` request.
2. [esp8266-display-example.yaml](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/esp8266-display-example.yaml): This example configuration should be used for a Soyosource without a display / with a display port. The ESP should be connected to the display port using a level shifter and to the RS485 port using a RS485 converter. I assume this is the solution you would like to use. The mentioned development board above is a good choice!
3. [esp8266-wifi-dongle-example.yaml](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/esp8266-wifi-dongle-example.yaml): This is the example to transform/convert your WiFi dongle into a ESPHome node using this project.

**Reply by Huy13031993** [reply]
Xin chào bạn. Tôi đọc bài thấy hay tôi có soyosource.  Có esp8266.  Tôi muốn hỏi thêm.  Tôi có thể liên hệ với bạn qua mail hay Facebook không. Mail của tôi là lehuy13031993@gmail.com.vn. Facebook của tôi là Lê Xuân Huy

**Reply by syssi** [reply]
Please create a new discussion and ask your questions here. I will provide some guidance!

**Reply by Huy13031993** [reply]
Có thể chỉ tôi cách đấu nối. Và cách xem trên điện thoại không.  Tôi chỉ biết nạp esp8266 qua aduino ide. Soyosource của tôi có màn hình.

**Reply by Huy13031993** [reply]
![Uploading 16797465496865167302834482804762.jpg…]()

**Reply by Huy13031993** [reply]
Bạn có ở đây không

**Reply by syssi** [reply]
Could you upload the photo another time? It didn't complete last time.

Do you have access to a linux host?

**Reply by Huy13031993** [reply]
I use Windows. Do you have an application to install on your android phone. Purpose To view and set parameters for soyosource via wifi esp8266.

**Reply by syssi** [reply]
Do I understand you correctly: You don't want to push some power demand to the inverter. You want to replace the display to retrieve the status and change the settings from remote?

**Reply by Huy13031993** [reply]
That's right. I just want to use it to view and set parameters via wifi. I already have a wifi module. But don't know how to connect. I see it has an rs485 port connecting to the meter. There is a 4-pin jack connected to the monitor.

**Reply by syssi** [reply]
Does these details help already? https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/tree/main/docs/display-version

**Reply by Huy13031993** [reply]
Right.  I want to see the status and settings remotely.  Via wifi module.

**Reply by syssi** [reply]
If you say "wifi module" do you mean a generic ESP or the OEM wifi dongle of the Soyosource WiFi version?

**Comment by MmeHulot** [comment]
I have a "silent" Soyo without Display but with Wifi-Dongle (Displayport). Power-demand-Value over RS485 with level shifter is already working, no problems :-).  I leave this as it is. I will ad the [esp8266-display-example.yaml](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/blob/main/esp8266-display-example.yaml) using the mentioned ESP-level-shifting-wifi-board to be able to "talk" to the Soyo. It's not really necessary but it's nice to have/monitor Soyos' data.

Thank you for answering my question!

**Reply by syssi** [reply]
If you use this project already (for passing the power-demand-value via RS485 to the Soyo) and your ESP has two free/unused GPIOs you could extend your setup by adding one of these [level shifter modules](https://de.aliexpress.com/item/4000552920569.html) to talk to the display port.

**Comment by Huy13031993** [comment]
Yes. I have 1 soyosource device with a display. I have esp8266. I see your post mentions tracking and installation. I want to do it
follow you. Can you guide me.

**Reply by syssi** [reply]
Did you build this setup already? https://github.com/syssi/esphome-soyosource-gtn-virtual-meter#schematics

**Reply by Huy13031993** [reply]
I don't have a conversion module. Rs458 to ttl. I see on inverter only 1 rs485 port. Using the connection with the load tracking meter. I don't know where to connect this module.

**Reply by Huy13031993** [reply]
Thank you for answering the question.

**Reply by syssi** [reply]
This project replaces the OEM limiter/meter with the current clamps. You have to use a smart meter, a shelly 3em or something else to measure your current power consumption. This project is able to transform the power demand into a Soyosource compatible RS485 message to instruct the inverter to produce as much power as needed.

**Reply by Huy13031993** [reply]
The inverter itself has a load tracking device. It measures the power consumption. And the inverter produces electricity according to the measured capacity from the metering device.

**Reply by syssi** [reply]
That's true. The `soyosource_virtual_meter` component is a drop-in replacement to control the outpot power of the inverter more accurate.

**Reply by Huy13031993** [reply]
Yes.  I understand what you mean.  I think get data from this rs485 port for remote monitoring and setting.
